pytoda/datasets/annotated_dataset.py

```diff
@@ -0,0 +1,92 @@
+"""Implementation of AnnotatedDataset class."""
+import torch
+import pandas as pd
+from torch.utils.data import Dataset
+from ..types import DrugSensitivityData
+
+
+class AnnotatedDataset(Dataset):
+    """
+    Annotated dataset implementation.
+    """
+
+    def __init__(
+        self,
+        annotations_filepath: str,
+        input_data: Dataset,
+        index_col: int,
+        device: torch.device = torch.
+        device('cuda' if torch.cuda.is_available() else 'cpu'),
+        backend: str = 'eager',
+        **kwargs
+    ) -> None:
+        """
+        Initialize an annotated dataset.
+        E.g. the  input_data could be SMILES and the annotations could be
+        single or multi task labels.
+
+        Args:
+            annotations_filepath (str): path to the annotations of a dataset
+                .csv file. Currently, the only supported format is .csv, the
+                last column should point to an ID that is also contained in
+                input_data.
+            input_data (Dataset): path to .smi file.
+            index_col (int): index column of annotations_filepath csv file
+            smiles_langśage (SMILESLanguage): a smiles language.
+                Defaults to None.
+            device (torch.device): device where the tensors are stored.
+                Defaults to gpu, if available.
+            backend (str): memeory management backend.
+                Defaults to eager, prefer speed over memory consumption.
+                Note that at the moment only the gene expression and the
+                smiles datasets implement both backends. The drug sensitivity
+                data are loaded in memory.
+            kwargs (dict): additional parameters for pd.read_csv.
+        """
+        Dataset.__init__(self)
+        self.annotations_filepath = annotations_filepath
+        # device
+        self.device = device
+        # backend
+        self.backend = backend
+        # e.g. SMILES
+        self.input_data = input_data
+
+        self.annotated_data_df = pd.read_csv(
```

Comment:
Here I would either include `index_col` in `kwargs` or make sure to add `_ = kwargs.pop('index_col', None)` before `pd.read_csv` to make sure we don't raise.

Reply:
I updated the default of `index_col` (which is `None`) to 0 (set in constructor)